### PR TITLE
[SEMI-MODULAR] Nerve staple bounty

### PR DESCRIPTION
**Pull request by Seris02**
## About The Pull Request

Nerve staple bounty

Adds a -10 value quirk which applies an item that pacifies you (-8) and permanently takes up the glasses slot (-2). **Can't be taken with nearsighted or pacifist quirks**.

Also fixes the ghost cafe interaction with quirks so that things spawn properly.

## How This Contributes To The Skyrat Roleplay Experience

bounty + bug fix

## Changelog

:cl:
add: Adds the nerve staple, a nodrop item that pacifies you, and the corresponding quirk
/:cl:

### CODE REVIEW

**Comment by Sonic121x** [comment]
can it be get rid of with the HARS mutation?(datum/mutation/human/headless)
or by amputate the head and reconnect it.

File: code/modules/awaymissions/corpse.dm [hunk 1]
```diff
@@ -281,9 +282,10 @@
 		//Pre-job equips so Voxes dont die
 		H.dna.species.pre_equip_species_outfit(null, H)
 		H.regenerate_icons()
-		SSquirks.AssignQuirks(H, user.client, TRUE, TRUE, null, FALSE, H)
-		for(var/datum/loadout_item/item as anything in loadout_list_to_datums(H?.client?.prefs?.loadout_list))
-			item.post_equip_item(H.client?.prefs, H)
+		if (!handles_loadout_and_quirks)
+			SSquirks.AssignQuirks(H, user.client, TRUE, TRUE, null, FALSE, H)
+			for(var/datum/loadout_item/item as anything in loadout_list_to_datums(H?.client?.prefs?.loadout_list))
+				item.post_equip_item(H.client?.prefs, H)
```
Comment (tf-4) [inline code/modules/awaymissions/corpse.dm]: skyrat edit

**Comment by SpaceVampire** [comment]
If you can just remove the item and the effect stops, I don't know if it is really a good replacement for those quirks at the exact same price. Given that there is normally no real way to deal with any of the other quirks. You can't just get brain surgery in the middle of the round to fix them or anything.

**Comment by tf-4** [comment]
Please don't bundle fixes in with features

**Comment by Gandalf2k15** [comment]
Make it so that if you remove it, they die

File: code/modules/awaymissions/corpse.dm (hunk 1 above)
Comment (Seris02) [reply code/modules/awaymissions/corpse.dm]: doesn't need one, it's already inside a skyrat edit

File: code/modules/awaymissions/corpse.dm [hunk 2]
```diff
@@ -230,6 +230,7 @@
 	var/chosen_alias
 	var/is_pref_char
 	var/last_ckey //For validation of the user
+	var/handles_loadout_and_quirks = FALSE
```
Comment (jjpark-kb) [inline code/modules/awaymissions/corpse.dm]: needs to have documentation ///

File: modular_skyrat/master_files/code/datums/quirks/negative.dm [hunk 3]
```diff
@@ -0,0 +1,22 @@
+/datum/quirk/equipping/nerve_staple
+	name = "Nerve Stapled"
+	desc = "You're a pacifist. Not because you want to be, but because of the device stapled into your eye."
+	value = -10 //pacifism = -8, losing eye slots = -2
+	gain_text = "<span class='danger'>You suddenly can't raise a hand to hurt others!</span>"
+	lose_text = "<span class='notice'>You think you can defend yourself again.</span>"
+	medical_record_text = "Patient is nerve stapled and is unable to harm others."
+	icon = "hand-peace"
+	forced_items = list(/obj/item/clothing/glasses/nerve_staple = list(ITEM_SLOT_EYES))
+	var/obj/item/clothing/glasses/nerve_staple/staple
```
Comment (jjpark-kb) [inline modular_skyrat/master_files/code/datums/quirks/negative.dm:11]: needs to have documentation ///

File: modular_skyrat/master_files/code/datums/quirks/neutral.dm [hunk 4]
```diff
@@ -27,7 +27,8 @@
 				if (success)
 					break
 		equipped_items[item] = success
-	on_equip_items(equipped_items)
+	for (var/obj/item/equipped as anything in equipped_items)
+		on_equip_item(equipped, equipped_items[equipped])
```
Comment (jjpark-kb) [inline modular_skyrat/master_files/code/datums/quirks/neutral.dm]: whats the difference between on_equip_items and on_equip_item... why would you rather have the singular and put a loop when I assume the plural does that.

File: modular_skyrat/master_files/code/datums/quirks/neutral.dm [hunk 5]
```diff
@@ -73,12 +74,14 @@
...
-	for (var/obj/item/clothing/accessory/breathing/acc in items_equipped)
-		acc.breath_type = breath_type
-		if (acc.can_attach_accessory(human_holder?.w_uniform))
-			acc.attach(human_holder.w_uniform, human_holder)
+	if (!istype(equipped, /obj/item/clothing/accessory/breathing))
+		return
+	var/obj/item/clothing/accessory/breathing/acc = equipped
+	acc.breath_type = breath_type
+	if (acc.can_attach_accessory(human_holder?.w_uniform))
+		acc.attach(human_holder.w_uniform, human_holder)
```
Comment (jjpark-kb) [inline modular_skyrat/master_files/code/datums/quirks/neutral.dm]: this does not seem related to nerve staples. atomize.

File: modular_skyrat/master_files/code/datums/quirks/neutral.dm [hunk 6]
```diff
@@ -119,12 +122,10 @@
...
 	var/mob/living/carbon/carbon_holder = quirk_holder
-	if (!istype(carbon_holder))
+	if (!successful || !istype(carbon_holder) || !istype(equipped, /obj/item/tank/internals))
 		return
-	for (var/obj/item/tank/internals/tank in items_equipped)
-		if (items_equipped[tank])
-			carbon_holder.internal = tank
-			carbon_holder.update_internals_hud_icon(1)
+	carbon_holder.internal = equipped
+	carbon_holder.update_internals_hud_icon(1)
```
Comment (jjpark-kb) [inline modular_skyrat/master_files/code/datums/quirks/neutral.dm]: this does not seem related to nerve staples. atomize.

**Review by jjpark-kb — CHANGES_REQUESTED** [review]
deal with what I mentioned in order to gain approval.

**Comment by Seris02** [comment]
do not merge until #10010 is merged and after that I'll fix up the code to reflect that PR

**Comment by tf-4** [comment]
fix changelog pls

File: modular_skyrat/master_files/code/datums/quirks/negative.dm [hunk 7]
```diff
@@ -0,0 +1,23 @@
+/datum/quirk/equipping/nerve_staple
+	name = "Nerve Stapled"
+	desc = "You're a pacifist. Not because you want to be, but because of the device stapled into your eye."
+	value = -10 //pacifism = -8, losing eye slots = -2
+	gain_text = "<span class='danger'>You suddenly can't raise a hand to hurt others!</span>"
+	lose_text = "<span class='notice'>You think you can defend yourself again.</span>"
```
Comment (tf-4) [inline modular_skyrat/master_files/code/datums/quirks/negative.dm]:
```suggestion
	gain_text = span_danger("You suddenly can't raise a hand to hurt others!")
	lose_text = span_notice("You think you can defend yourself again.")
```
Yes, this does work

**Review by tf-4 — COMMENTED** [review]
Final change then ready to go

File: modular_skyrat/master_files/code/datums/quirks/negative.dm [hunk 8]
```diff
@@ -0,0 +1,23 @@
...
+/datum/quirk/equipping/nerve_staple/on_equip_item(obj/item/equipped, successful)
+	if (!istype(equipped, /obj/item/clothing/glasses/nerve_staple))
+		return
+	staple = equipped
+
+/datum/quirk/equipping/nerve_staple/remove()
+	. = ..()
+	if (!staple || staple != quirk_holder.get_item_by_slot(ITEM_SLOT_EYES))
+		return
+	to_chat(quirk_holder, span_warning("The nerve staple suddenly melts off your face!"))
```
Comment (Fikou) [inline modular_skyrat/master_files/code/datums/quirks/negative.dm]: this makes it sound like your face is being melted

File: modular_skyrat/master_files/code/modules/clothing/glasses/nerve_staple.dm [hunk 9]
```diff
@@ -0,0 +1,24 @@
+/obj/item/clothing/glasses/nerve_staple
```
Comment (Fikou) [inline modular_skyrat/master_files/code/modules/clothing/glasses/nerve_staple.dm:1]: make it acid/fireproof?

**Review by tf-4 — APPROVED** [review]
Seems good

**Pull request merged**